### PR TITLE
SemVer Version format support

### PR DESCRIPTION
**Pull request by shresthagrawal**
rebased and cherrypicked into a fresh branch

### CODE REVIEW

**Comment by jpopelka** [comment]
[test]

File: release_bot/releasebot.py [hunk 1]
```diff
@@ -66,13 +66,44 @@ def load_release_conf(self):
...
         self.new_release.update(release_conf)
 
+    def process_version_from_title(self, title, latest_version):
+        """
+        checks for a valid version keyword, and if found
+        returns the requested release version
+        :param title: the pr/issue title to check keyword for
+        :param latest_version: the current latest version with type Version()
+        :return match: true if it is a valid version
+        :return version: version string for requested latest release
```
Comment (TomasTomecek) [inline release_bot/releasebot.py]: codacy complains about trailing whitespace here, please remove it

Comment (shresthagrawal) [reply release_bot/releasebot.py]: okay! 
for the Test do I need to write a test that involves complete process from creating an a new issue named for eg: 'new major release' to checking whether the bot detects it does the release or I can just write a unit test for the specific funtion where I just manually send the current version and possible keywords and check for the results ? I think the former is better ?

**Comment by shresthagrawal** [comment]
@TomasTomecek why does Travic-Ci fails on a random occasion. :>>

**Comment by rpitonak** [comment]
> @TomasTomecek why does Travic-Ci fails on a random occasion. :>>

I retriggered the build, it wasn't related to this PR.

File: release_bot/releasebot.py (hunk 1 above)
Comment (TomasTomecek) [reply release_bot/releasebot.py]: no need to write an end to end test. Just write a test where you pass several sets of inputs to the method process_version_from_title and check if you are getting the results you expect -- unit test is just fine

**Comment by rpitonak** [comment]
@shresthagrawal [Codacy is complaining](https://app.codacy.com/app/user-cont/release-bot/pullRequest?prid=3227955) about unused imports, can you fix it? Otherwise LGTM. Nicely done!

**Pull request merged**